test/factories/data_array.rb

```diff
@@ -9,10 +9,25 @@
       # convenience allowing specification of a Gene, rather than id and name individually
       gene { nil }
     end
+    transient do
+      # convenience allowing specification of a CellMetadatum, rather than id and name individually
+      cell_metadatum { nil }
+    end
+
     study { study_file.study }
     cluster_name { cluster_group.try(:name) || study_file.try(:name) }
     cluster_group_id { cluster_group.try(:id) }
-    linear_data_id { cluster_group.try(:id) || gene.try(:id) }
-    linear_data_type { cluster_group_id ? 'ClusterGroup' : 'Gene' }
+    linear_data_id { cluster_group.try(:id) || gene.try(:id) || cell_metadatum.try(:id) || study.id }
```

Comment:
Nice - I was struggling with this exact issue and I think this encapsulates it nicely.